File: stageTemplates/src/app/models/role.ts
export class Role {
    id: number;
    designation: string;
    creer: boolean;
    modifier: boolean;
    supprimer: boolean;
    lecture: boolean;
    controlTotal: boolean=false;
  
    constructor(
      id: number,
      designation: string,
      creer: boolean,
      modifier: boolean,
      supprimer: boolean,
      lecture: boolean
    ) {
      this.id = id;
      this.designation = designation;
      this.creer = creer;
      this.modifier = modifier;
      this.supprimer = supprimer;
      this.lecture = lecture;
    }
  }
  

File: CondorV/CondorV/Models/BD/Role.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace CondorV.Models.BD
{
    public class Role
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }
        public string Designation { get; set; }
        public bool Creer { get; set; } =false;
        public bool Modifier { get; set; } =false;
        public bool Supprimer { get; set; } =false;
        public bool Lecture { get; set; } =false;
        [JsonIgnore]
        public ICollection<Utilisateur>? Utilisateurs { get; set; }
        public Role() { }
        public Role(string designation)
        {
          Designation=designation;
        }



    }
}
